<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
	<title>MySQL++ Reference Manual</title>
	<meta http-equiv="Content-type" content="text/html;charset=iso-8859-1">
	<link href="doxygen.css" rel="stylesheet" type="text/css">
	<link href="refman.css" rel="stylesheet" type="text/css">
</head>
<body>
<!-- Generated by Doxygen 1.4.7 -->
<div class="tabs">
  <ul>
    <li><a href="index.html"><span>Main&nbsp;Page</span></a></li>
    <li><a href="namespaces.html"><span>Namespaces</span></a></li>
    <li id="current"><a href="annotated.html"><span>Classes</span></a></li>
    <li><a href="files.html"><span>Files</span></a></li>
  </ul></div>
<div class="tabs">
  <ul>
    <li><a href="annotated.html"><span>Class&nbsp;List</span></a></li>
    <li><a href="hierarchy.html"><span>Class&nbsp;Hierarchy</span></a></li>
    <li><a href="functions.html"><span>Class&nbsp;Members</span></a></li>
  </ul></div>
<div class="nav">
<b>mysqlpp</b>::<a class="el" href="classmysqlpp_1_1ConnectionFailed.html">ConnectionFailed</a></div>
<h1>mysqlpp::ConnectionFailed Class Reference</h1><!-- doxytag: class="mysqlpp::ConnectionFailed" --><!-- doxytag: inherits="mysqlpp::Exception" --><a class="el" href="classmysqlpp_1_1Exception.html">Exception</a> thrown when there is a problem related to the database server connection.  
<a href="#_details">More...</a>
<p>
<code>#include &lt;<a class="el" href="exceptions_8h-source.html">exceptions.h</a>&gt;</code>
<p>
Inheritance diagram for mysqlpp::ConnectionFailed:<p><center><img src="classmysqlpp_1_1ConnectionFailed__inherit__graph.png" border="0" usemap="#mysqlpp_1_1ConnectionFailed__inherit__map" alt="Inheritance graph"></center>
<map name="mysqlpp_1_1ConnectionFailed__inherit__map">
<area href="classmysqlpp_1_1Exception.html" shape="rect" coords="29,5,181,32" alt="">
</map>
<center><font size="2">[<a href="graph_legend.html">legend</a>]</font></center>Collaboration diagram for mysqlpp::ConnectionFailed:<p><center><img src="classmysqlpp_1_1ConnectionFailed__coll__graph.png" border="0" usemap="#mysqlpp_1_1ConnectionFailed__coll__map" alt="Collaboration graph"></center>
<map name="mysqlpp_1_1ConnectionFailed__coll__map">
<area href="classmysqlpp_1_1Exception.html" shape="rect" coords="29,96,181,123" alt="">
</map>
<center><font size="2">[<a href="graph_legend.html">legend</a>]</font></center><a href="classmysqlpp_1_1ConnectionFailed-members.html">List of all members.</a><table border="0" cellpadding="0" cellspacing="0">
<tr><td></td></tr>
<tr><td colspan="2"><br><h2>Public Member Functions</h2></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top">&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmysqlpp_1_1ConnectionFailed.html#bede9f71e0283af57a7101c0137eaaab">ConnectionFailed</a> (const char *w=&quot;&quot;, int e=0)</td></tr>

<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Create exception object.  <a href="#bede9f71e0283af57a7101c0137eaaab"></a><br></td></tr>
<tr><td class="memItemLeft" nowrap align="right" valign="top">int&nbsp;</td><td class="memItemRight" valign="bottom"><a class="el" href="classmysqlpp_1_1ConnectionFailed.html#ce3fd471b5cb3e92812c9a82758ec966">errnum</a> () const </td></tr>

<tr><td class="mdescLeft">&nbsp;</td><td class="mdescRight">Return the error number corresponding to the error message returned by <a class="el" href="classmysqlpp_1_1Exception.html#970fe1876f16ea720382dff87f80cf17">what()</a>, if any.  <a href="#ce3fd471b5cb3e92812c9a82758ec966"></a><br></td></tr>
</table>
<hr><a name="_details"></a><h2>Detailed Description</h2>
<a class="el" href="classmysqlpp_1_1Exception.html">Exception</a> thrown when there is a problem related to the database server connection. 
<p>
This is thrown not just on making the connection, but also on shutdown and when calling certain of Connection's methods that require a connection when there isn't one. 
<p>
<hr><h2>Constructor &amp; Destructor Documentation</h2>
<a class="anchor" name="bede9f71e0283af57a7101c0137eaaab"></a><!-- doxytag: member="mysqlpp::ConnectionFailed::ConnectionFailed" ref="bede9f71e0283af57a7101c0137eaaab" args="(const char *w=&quot;&quot;, int e=0)" -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">mysqlpp::ConnectionFailed::ConnectionFailed           </td>
          <td>(</td>
          <td class="paramtype">const char *&nbsp;</td>
          <td class="paramname"> <em>w</em> = <code>&quot;&quot;</code>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">int&nbsp;</td>
          <td class="paramname"> <em>e</em> = <code>0</code></td><td>&nbsp;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td><td width="100%"><code> [inline, explicit]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Create exception object. 
<p>
<dl compact><dt><b>Parameters:</b></dt><dd>
  <table border="0" cellspacing="2" cellpadding="0">
    <tr><td valign="top"></td><td valign="top"><em>w</em>&nbsp;</td><td>explanation for why the exception was thrown </td></tr>
    <tr><td valign="top"></td><td valign="top"><em>e</em>&nbsp;</td><td>the error number from the underlying database API </td></tr>
  </table>
</dl>

</div>
</div><p>
<hr><h2>Member Function Documentation</h2>
<a class="anchor" name="ce3fd471b5cb3e92812c9a82758ec966"></a><!-- doxytag: member="mysqlpp::ConnectionFailed::errnum" ref="ce3fd471b5cb3e92812c9a82758ec966" args="() const " -->
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">int mysqlpp::ConnectionFailed::errnum           </td>
          <td>(</td>
          <td class="paramname">          </td>
          <td>&nbsp;)&nbsp;</td>
          <td width="100%"> const<code> [inline]</code></td>
        </tr>
      </table>
</div>
<div class="memdoc">

<p>
Return the error number corresponding to the error message returned by <a class="el" href="classmysqlpp_1_1Exception.html#970fe1876f16ea720382dff87f80cf17">what()</a>, if any. 
<p>
If the error number is 0, it means that the error message doesn't come from the underlying database API, but rather from MySQL++ itself. This happens when an error condition is detected up at this higher level instead of letting the underlying database API do it. 
</div>
</div><p>
<hr>The documentation for this class was generated from the following file:<ul>
<li><a class="el" href="exceptions_8h-source.html">exceptions.h</a></ul>
<hr size="1"><address style="align: right;"><small>Generated on 10 Dec 2013 for MySQL++ by&nbsp;
<a href="http://www.doxygen.org/index.html">
<img src="doxygen.png" alt="doxygen" align="middle" border="0"></a> 1.4.7 </small></address>
</body>
</html>
